tests/backends/storage/test_swift.py

```diff
@@ -0,0 +1,286 @@
+
+FREEZED_NOW = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+SWIFT_OPTIONS = [
+    "os_tenant_id",
+    "os_tenant_name",
+    "os_username",
+    "os_password",
+    "os_region_name",
+    "os_storage_url",
+    "os_user_domain_name",
+    "os_project_domain_name",
+    "os_auth_url",
+    "os_identity_api_version",
+]
+HISTORY = [
+    {
+        "backend": "swift",
+        "command": "fetch",
+        "id": "2020-04-29.gz",
+        "filename": "2020-04-29.gz",
+        "size": 23424233,
+        "fetched_at": "2020-10-07T16:40:25.887664+00:00",
+    },
+    {
+        "backend": "swift",
+        "command": "fetch",
+        "id": "2020-04-30.gz",
+        "filename": "2020-04-30.gz",
+        "size": 23424233,
+        "fetched_at": "2020-10-07T16:37:25.887664+00:00",
+    },
+]
+LISTING = [
+    {
+        "bytes": 23424233,
+        "last_modified": "2021-01-20T10:41:03.950030",
+        "hash": "01018bc73b8b3491c6c3d543044a702b",
+        "name": "2020-04-29.gz",
+        "content_type": "application/gzip",
+    },
+    {
+        "bytes": 23424233,
+        "last_modified": "2021-01-22T13:53:58.751680",
+        "hash": "b9f999739e55ab42ad852453d145a6a9",
+        "name": "2020-04-30.gz",
+        "content_type": "text/plain",
+    },
+    {
+        "bytes": 23424233,
+        "last_modified": "2021-01-22T13:53:03.541030",
+        "hash": "b9f999739e55ab42ad852453d145a6a9",
+        "name": "2020-05-01.gz",
+        "content_type": "text/plain",
+    },
+]
+
+
+def list_page(success=True):
+    """Mocks the output of SwiftService.list"""
+    page = {"success": success, "container": "ralph_logs_container"}
+    listing = {"listing": LISTING}
+    error = {"error": "Container not found"}
+    page.update(listing if success else error)
+    return page
+
+
+@pytest.mark.parametrize("option", SWIFT_OPTIONS)
+@pytest.mark.parametrize(
+    "value", [None, inspect._empty()]  # pylint: disable=protected-access
+)
+def test_given_none_or_empty_option_swift_storage_should_raise_exception(
+    option, value, swift
+):
+    """Check that SwiftStorage raises BackendParameterException for None or empty options"""
+
+    options = swift.options
+    options[option] = value
+    msg = f"SwiftStorage backend instance requires the {option} option to be set"
+    with pytest.raises(BackendParameterException, match=msg):
+        SwiftStorage(**options)
+
+
+@pytest.mark.parametrize("exception", [SwiftError, SwiftError])
+def test_throws_swift_error_decorator_should_re_raise_exception(exception):
+    """Check that when a function is decorated with swift_error_decorator
+    it catches, logs and re-raises SwiftError or ClientException
+    """
+
+    msg = f"Error: {exception.__class__}"
+
+    @throws_swift_error
+    def will_throw_exception():
+        """Throws an instance of exception"""
+        raise exception(msg)
+
+    with TestCase().assertLogs("ralph", level="ERROR") as cm:
```

Comment:
You can use the `caplog` pytest fixture instead. But this test will be removed if we remove tested decorator.

https://docs.pytest.org/en/stable/logging.html#caplog-fixture

Reply:
Good point) Wasn't aware of caplog, thanks)